File: cpp_solutions/2515. Shortest Distance to Target String in a Circular Array.cpp
class Solution {
public:
    void print_vec(vector<string> t) {
        for(auto it : t) {
            cout << it << " ";
        }
    }
    int closetTarget(vector<string>& words, string target, int startIndex) {
        vector<string> temp;
        int n = words.size();
        for(auto it : words){
            temp.emplace_back(it);
        }
        for(auto it : words){
            temp.emplace_back(it);
        }
        for(auto it : words){
            temp.emplace_back(it);
        }
        
        int start = n + startIndex;
        // move left and right to check
        int ans = 100000;
        for(int i = start; i < temp.size(); ++i) {
            if (temp[i] == target){
                ans = min(ans, i - start);
            }
        }

        for(int i = start; i >= 0; --i) {
            if (temp[i] == target){
                ans = min(ans, start - i);
            }
        }

        if(ans!=100000){
            return ans;
        } else {
            return -1;
        }
    }
};

File: python3_solution_set2/2515. Shortest Distance to Target String in a Circular Array.py
class Solution:
    def closetTarget(self, words: List[str], target: str, startIndex: int) -> int:

        words = words + words + words

        start = len(words)//3 + startIndex
        
        ## move in left and right direction to find the target words

        ans = float('inf')

        for x in range(start, len(words), +1):
            if words[x] == target:
                ans = min(ans, x-start)

        for x in range(start,0,-1):
            if words[x] == target:
                ans = min(ans, start-x)

        return ans if ans!=float('inf') else -1
